code/modules/mapping/ruins.dm

```diff
@@ -11,8 +11,15 @@
 
 		for(var/turf/check in get_affected_turfs(central_turf,1))
 			var/area/new_area = get_area(check)
-			if(!(istype(new_area, allowed_areas)) || check.flags_1 & NO_RUINS_1)
+			if(check.flags_1 & NO_RUINS_1)
 				valid = FALSE
+			else
+				for(var/type in allowed_areas)
+					valid = FALSE // set to false before we check
+					if(istype(new_area, type)) // it's at least one of our types so it's whitelisted
+						valid = TRUE
+						break
+			if(!valid)

```

Comment:
you could move the valid = FALSE assignment to outside the loop no need to reassign each time with the break